lib/datasource/gitlab-releases/index.ts

```diff
@@ -0,0 +1,55 @@
+import { cache } from '../../util/cache/package/decorator';
+import { GitlabHttp } from '../../util/http/gitlab';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import type { GitlabRelease } from './types';
+
+export class GitlabReleasesDatasource extends Datasource {
+  static readonly id = 'gitlab-releases';
+
+  readonly defaultRegistryUrls = ['https://gitlab.com'];
+
+  static readonly registryStrategy = 'first';
+
+  http = new GitlabHttp();
```

Comment:
I think we need to add a new constructor to override the hostType, see #11136 for github sample